src/data-fetcher-loop/data-fetcher-loop.ts

```diff
@@ -77,32 +75,33 @@ export const runDataFetcher = async () => {
     const signedDataFetchIntervalMs = signedDataFetchInterval * 1000;
 
     // Better to log the non-decomposed object to see which URL comes from which chain-provider group.
-    logger.debug('Fetching data from signed APIs.', { signedApiUrls });
+    logger.debug('Signed API URLs.', { signedApiUrls });
     const urls = uniq(
       Object.values(signedApiUrls)
         .map((urlsPerProvider) => Object.values(urlsPerProvider))
         .flat(2)
     );
 
+    const urlCount = urls.length;
+    const staggerTimeMs = signedDataFetchIntervalMs / urlCount;
+    logger.info('Fetching signed data', { urlCount, staggerTimeMs });
     const fetchResults = await Promise.all(
-      urls.map(async (url) =>
-        go(
-          async () => {
-            const signedDataApiResponse = await callSignedApi(
-              url,
-              Math.ceil(signedDataFetchIntervalMs * HTTP_SIGNED_DATA_API_TIMEOUT_MULTIPLIER)
-            );
-            if (!signedDataApiResponse) return;
-
-            for (const signedData of signedDataApiResponse) {
-              saveSignedData(signedData);
-            }
-          },
-          { totalTimeoutMs: signedDataFetchIntervalMs }
-        )
-      )
+      urls.map(async (url, index) => {
+        await sleep(staggerTimeMs * index);
+
+        // NOTE: We allow each Signed API call to take full signedDataFetchIntervalMs. Because these calls are
+        // staggered, it means that there can be pending requests from different data fetcher loops happening at the
+        // same time. This does not matter much, because we only save the freshest signed data.
+        const signedDataApiResponse = await callSignedApi(url, signedDataFetchIntervalMs);
```

Comment:
If we decrease `signedDataFetchIntervalMs` to value like 2.5 or 5 secs, it can cause issues but otherwise (selecting it to be greater than 10 secs) it is feasible.